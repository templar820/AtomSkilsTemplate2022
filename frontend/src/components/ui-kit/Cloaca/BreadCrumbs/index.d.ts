import * as React from 'react';
import { EmotionStylesType } from 'lib-ui';

export interface BreadCrumbsProps {
    /** size of entire component */
    size?: 'sm' | 'lg';
    /** items to show, example: [{title: 'test', link: '#'}] */
    items: Array<{ title: string; link: string }>;
    /** color to highlighted (last) component */
    lastItemColor?: string;
    /** Callback fire then not disabled link is clicked, func (item) */
    onLinkClick?: (item: React.ReactElement) => void;
    /** render function to redefine link component */
    linkComponent?: React.ReactElement;
    /** Style group: outer element*/
    wrapStyle?: EmotionStylesType;
    /** Style group: each link or text */
    itemStyle?: EmotionStylesType;
    /** Style group: icons between links */
    iconStyle?: EmotionStylesType;
    /** Define bounds of component */
    width?: string;
    /** Define bounds of component */
    height?: string;
}

declare const BreadCrumbs: React.ComponentType<BreadCrumbsProps>;

export default BreadCrumbs;
