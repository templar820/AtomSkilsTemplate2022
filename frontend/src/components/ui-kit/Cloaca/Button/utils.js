import { lighten } from 'lib-root/utils/styleMixins';

const btnSizes = () => {
    return {
        lg: {
            iconBtnWidth: '60px',
            iconSize: '32px',
            vPadding: '30px',
            height: '60px'
        },
        md: {
            iconBtnWidth: '48px',
            iconSize: '32px',
            vPadding: '24px',
            height: '48px'
        },
        sm: {
            iconBtnWidth: '32px',
            iconSize: '16px',
            vPadding: '16px',
            height: '32px'
        },
        xs: {
            iconBtnWidth: '24px',
            iconSize: '16px',
            height: '24px'
        }
    };
};

const spinnerModes = {
    primary: 'secondary',
    secondary: 'primary',
    white: 'default',
    outline: 'default'
};

const _lighten = (color, { colorDiff = 20 }) => {
    return lighten(color, colorDiff);
};

export { btnSizes, _lighten, spinnerModes };
