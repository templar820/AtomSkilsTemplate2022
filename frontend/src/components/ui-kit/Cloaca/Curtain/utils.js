import { useCallback, useEffect, useRef, useState } from 'react';
import config from 'lib-root/components/Curtain/config';

const callFn = (fn, size) => {
    if (typeof fn === 'function') fn({ size });
};
const useDragUpdater = (onDragStart, onDragEnd, curtainRef, vertical) => {
    const [isDragging, _setIsDragging] = useState(false);
    const setIsDragging = useCallback(
        (payload) => {
            const size = curtainRef.current.style[vertical ? 'height' : 'width'];
            callFn(payload ? onDragStart : onDragEnd, size);
            _setIsDragging(payload);
        },
        [onDragStart, onDragEnd, vertical]
    );
    return [isDragging, setIsDragging];
};

function useDragHandler(position, max, min, curtainRef, setSize, onDragStart, onDragEnd) {
    const { vertical, inverse } = config.position[position];
    const [isDragging, setIsDragging] = useDragUpdater(onDragStart, onDragEnd, curtainRef, vertical);
    const mouseListener = useCallback(
        ({ type, clientX: x, clientY: y }) => {
            const size = calcPercent(vertical ? y : x, curtainRef, max, min, vertical, inverse) + '%';
            if (type === 'mousemove') {
                setSize(size);
            } else setIsDragging(false);
        },
        [position, max, min, setSize]
    );

    useEffect(() => {
        if (isDragging) {
            document.addEventListener('mousemove', mouseListener);
            document.addEventListener('mouseup', mouseListener);
        }
        return () => {
            document.removeEventListener('mousemove', mouseListener);
            document.removeEventListener('mouseup', mouseListener);
        };
    }, [isDragging]);

    return [isDragging, setIsDragging];
}

function trimValue(value, min, max) {
    if (value < min) return min;
    return value > max ? max : value;
}

function calcPercent(mousePosition, curtainRef, max, min, vertical = true, inverse = true) {
    const { left, top, width, height } = curtainRef.current.parentElement.getBoundingClientRect();
    const containerPosition = vertical ? top : left;
    const containerSide = vertical ? height : width;

    const percent = trimValue(((mousePosition - containerPosition) * 100) / containerSide, 0, 100);
    return trimValue(inverse ? 100 - percent : percent, min, max);
}

const useCurtainRef = (forwardedRef) => {
    const ref = useRef();
    const setRef = useCallback(
        (element) => {
            ref.current = element;
            switch (typeof forwardedRef) {
                case 'object':
                    if (forwardedRef) forwardedRef.current = element;
                    break;
                case 'function':
                    forwardedRef(element);
            }
        },
        [forwardedRef]
    );
    return [ref, setRef];
};

export { trimValue, calcPercent, useDragHandler, useCurtainRef };
