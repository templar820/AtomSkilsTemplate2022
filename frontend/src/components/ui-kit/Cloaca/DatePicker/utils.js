import {
    addMonths,
    areIntervalsOverlapping,
    startOfMonth,
    endOfMonth,
    startOfWeek,
    endOfWeek,
    getYear,
    getMonth,
    getDate,
    setYear,
    setMonth,
    setDate
} from 'date-fns';

const calcFocusDate = function(currentFocusedDate, props) {
    const { shownDate, date, months, ranges, focusedRange, displayMode } = props;
    // find primary date according the props
    let targetInterval;
    if (displayMode === 'dateRange') {
        const range = ranges[focusedRange[0]] || {};
        targetInterval = {
            start: range.startDate,
            end: range.endDate
        };
    } else {
        targetInterval = {
            start: date,
            end: date
        };
    }
    targetInterval.start = startOfMonth(targetInterval.start || new Date());
    targetInterval.end = endOfMonth(targetInterval.end || targetInterval.start);
    const targetDate = targetInterval.start || targetInterval.end || shownDate || new Date();

    // initial focus
    if (!currentFocusedDate) return shownDate || targetDate;

    // // just return targetDate for native scrolled calendars
    // if (props.scroll.enabled) return targetDate;
    const currentFocusInterval = {
        start: startOfMonth(currentFocusedDate),
        end: endOfMonth(addMonths(currentFocusedDate, months - 1))
    };
    if (areIntervalsOverlapping(targetInterval, currentFocusInterval)) {
        // don't change focused if new selection in view area
        return currentFocusedDate;
    }
    return targetDate;
};

const findNextRangeIndex = function(ranges, currentRangeIndex = -1) {
    const nextIndex = ranges.findIndex(
        (range, i) => i > currentRangeIndex && range.autoFocus !== false && !range.disabled
    );
    if (nextIndex !== -1) return nextIndex;
    return ranges.findIndex((range) => range.autoFocus !== false && !range.disabled);
};

const getMonthDisplayRange = function(date, dateOptions) {
    const startDateOfMonth = startOfMonth(date, dateOptions);
    const endDateOfMonth = endOfMonth(date, dateOptions);
    const startDateOfCalendar = startOfWeek(startDateOfMonth, dateOptions);
    const endDateOfCalendar = endOfWeek(endDateOfMonth, dateOptions);
    return {
        start: startDateOfCalendar,
        end: endDateOfCalendar,
        startDateOfMonth,
        endDateOfMonth
    };
};

const setNewDate = (value, newValue) => {
    if (!value) {
        return newValue;
    }
    const year = getYear(newValue);
    const month = getMonth(newValue);
    const day = getDate(newValue);
    const withYear = setYear(value, year);
    const withMonth = setMonth(withYear, month);
    const withDay = setDate(withMonth, day);
    return new Date(withDay);
};

export { calcFocusDate, findNextRangeIndex, getMonthDisplayRange, setNewDate };
