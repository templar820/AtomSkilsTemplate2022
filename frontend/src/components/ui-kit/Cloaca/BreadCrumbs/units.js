import styled from '@emotion/styled';

import { rem } from 'lib-root/utils/units';
import { hexToRGBA } from 'lib-root/utils/styleMixins';
import Icon from 'lib-ui/InlineIcons';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const Wrapper = styled.div`
    position: relative;
    flex: 0 0 ${rem(22)};
    margin-right: ${rem(8)};
    width: ${({ width }) => width};
    height: ${({ height }) => height};
    cursor: default;
    font-size: ${({ size }) => (size === 'sm' ? '16px' : '20px')};
    font-weight: 500;
    ${({ wrapStyle, ...restProps }) => applyEmotionStyle(wrapStyle, restProps)};
`;

const StyledLink = styled.a`
    margin-right: 5px;
    color: ${({ disabled, colors }) => (disabled ? `${hexToRGBA(colors.GrayScale_700, 0.5)}  !important` : 'inherit')};
    pointer-events: ${({ disabled }) => (disabled ? 'none !important' : 'auto')};
    user-select: ${({ disabled }) => (disabled ? 'none !important' : 'auto')};
    cursor: pointer;
    &:hover {
        text-decoration: none;
        color: inherit;
    }
`;
const StyledText = styled.span`
    margin-right: 5px;
    color: ${({ lastItemColor }) => lastItemColor};
    ${({ textStyles, ...restProps }) => applyEmotionStyle(textStyles, restProps)};
`;

const StyledItem = styled.span`
    margin-right: 5px;
    display: inline-flex;
    align-items: center;
    color: inherit;
    ${({ itemStyle, ...restProps }) => applyEmotionStyle(itemStyle, restProps)};
`;

const StyledIcon = styled(Icon)`
    margin-right: 5px;
    transform: rotate(270deg);
    ${({ iconStyle, ...restProps }) => applyEmotionStyle(iconStyle, restProps)};
`;

export { StyledIcon, StyledItem, StyledLink, StyledText, Wrapper };
