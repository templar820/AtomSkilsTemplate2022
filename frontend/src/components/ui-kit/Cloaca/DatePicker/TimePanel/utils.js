import { debounce } from 'lib-root/utils';

const FOCUSED_ELEMENT_NAME = ['prevArrow', 'nextArrow', 'hours', 'minutes', 'seconds'];

const timePanelElementChoiceFunc = ({ elementNum, setElementNum }) => ({
    event,
    focusedPicker,
    onArrowClick,
    timeValues,
    visiblePickers,
    prev,
    next
}) => {
    if (event.nativeEvent.code === 'ArrowRight' && elementNum <= 3) {
        debounce(() => setElementNum(elementNum + 1), 50)();
    }
    if (event.nativeEvent.code === 'ArrowLeft' && elementNum >= 1) {
        debounce(() => setElementNum(elementNum - 1), 50)();
    }
    if (event.nativeEvent.code === 'Enter') {
        prev && focusedPicker > 0 && onArrowClick({ direction: 'left' });

        next && focusedPicker < timeValues.length - visiblePickers && onArrowClick({ direction: 'right' });
    }
};

const onFocusElementFunc = ({ setElementNum }) => ({ elNum }) => {
    setElementNum(elNum);
};

export { timePanelElementChoiceFunc, onFocusElementFunc, FOCUSED_ELEMENT_NAME };
