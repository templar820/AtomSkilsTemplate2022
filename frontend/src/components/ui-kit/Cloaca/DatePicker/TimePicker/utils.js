import { useCallback, useMemo } from 'react';
import { getHours, getMinutes, getSeconds } from 'date-fns';
import { methodsDict } from './config';
import { debounce } from 'lib-root/utils';

// TODO: hooks naming
const ParseDateToTimeObject = ({ value }) =>
    useMemo(() => {
        if (!value) return {};
        return { hours: getHours(value), minutes: getMinutes(value), seconds: getSeconds(value) };
    }, [value]);

// TODO: hooks naming
const OnTimeItemClick = ({ method, currentValue, setValue, isControlled, callback }) =>
    useCallback(
        ({ newValue, onStateChange = () => {} }) => {
            setValue((prevValue) => {
                const newState = methodsDict[method](prevValue, newValue);
                callback && callback({ time: newState });
                onStateChange && onStateChange();
                return !isControlled ? newState : prevValue;
            });
        },
        [method, currentValue, isControlled, callback]
    );

// TODO: hooks naming
const GetTimeItems = (length) => useMemo(() => new Array(length).fill(1).map((_, index) => index), [length]);

const sumCounterChange = debounce(({ onItemClick, activeItem }) => {
    onItemClick({ newValue: activeItem + sumCounter, onStateChange: () => (sumCounter = 0) });
}, 50);

let sumCounter = 0;

const timePanelKeyControlFunc = ({ onClickOk }) => ({ event, activeItem, onItemClick }) => {
    if (event.nativeEvent.code === 'ArrowDown') {
        event.preventDefault();
        sumCounter++;
        sumCounterChange({ onItemClick, activeItem });
    }
    if (event.nativeEvent.code === 'ArrowUp') {
        event.preventDefault();
        sumCounter--;
        sumCounterChange({ onItemClick, activeItem });
    }

    if (event.nativeEvent.code === 'Enter') {
        debounce(() => onClickOk(), 150)();
    }
};

export {
    timePanelKeyControlFunc,
    ParseDateToTimeObject as parseDateToTimeObject,
    OnTimeItemClick as onTimeItemClick,
    GetTimeItems as getTimeItems
};
