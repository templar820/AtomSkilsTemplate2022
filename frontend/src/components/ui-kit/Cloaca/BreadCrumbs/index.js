import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import { rem } from 'lib-root/utils/units';
import { ColorsContext } from 'lib-root/colors';

import { StyledIcon, StyledItem, StyledLink, StyledText, Wrapper } from './units';
import { setClassName } from 'lib-root/utils/styleMixins';

/**
 * Компонент `BreadCrumbs` используется для создания цепочки навигации по каким-либо объектам.
 */
const BreadCrumbs = React.forwardRef(
    ({ items, onLinkClick, linkComponent, lastItemColor, itemStyle, iconStyle, ...rest }, ref) => {
        const colors = useContext(ColorsContext);
        lastItemColor = colors[lastItemColor] || lastItemColor;
        const lastItemIndex = items.length - 1;

        return (
            <Wrapper {...{ ref, ...rest }} className={setClassName({ props: rest, name: 'bread-crumbs' })}>
                {items.map((item, i) => (
                    <StyledItem
                        key={'breadC_' + i}
                        fontSize={rem(12)}
                        color={item.link && i !== lastItemIndex ? lastItemColor : colors.GrayScale_700}
                        {...{ itemStyle }}
                        className={setClassName({ props: rest, name: 'bread-crumbs__item' })}>
                        {i !== lastItemIndex ? (
                            <React.Fragment>
                                {linkComponent ? (
                                    React.cloneElement(linkComponent, { ...item })
                                ) : (
                                    <StyledLink {...{ colors }} disabled={!item.link} onClick={() => onLinkClick(item)}>
                                        {item.title}
                                    </StyledLink>
                                )}
                            </React.Fragment>
                        ) : (
                            <StyledText fontSize={rem(12)} {...{ lastItemColor }}>
                                {item.title}
                            </StyledText>
                        )}
                        {i !== lastItemIndex && (
                            <StyledIcon
                                icon={'Arrow_type_1'}
                                color={i === lastItemIndex - 1 ? lastItemColor : colors.GrayScale_700}
                                w={'24px'}
                                h={'24px'}
                                disabled={!item.link}
                                {...{ iconStyle }}
                            />
                        )}
                    </StyledItem>
                ))}
            </Wrapper>
        );
    }
);

BreadCrumbs.displayName = 'BreadCrumbs';
BreadCrumbs.propTypes = {
    /** size of entire component */
    size: PropTypes.oneOf(['sm', 'lg']),
    /** items to show, example: [{title: 'test', link: '#'}] */
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    /** color to highlighted (last) component */
    lastItemColor: PropTypes.string,
    /** Callback fire then not disabled link is clicked, func (item) */
    onLinkClick: PropTypes.func,
    /** render function to redefine link component */
    linkComponent: PropTypes.element,
    /** Style group: outer element*/
    wrapStyle: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Style group: each link or text */
    itemStyle: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Style group: icons between links */
    iconStyle: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Define bounds of component */
    width: PropTypes.string,
    /** Define bounds of component */
    height: PropTypes.string
};

BreadCrumbs.defaultProps = {
    items: [{ link: '#', title: '' }],
    size: 'sm',
    lastItemColor: 'secondary'
};

export default BreadCrumbs;
