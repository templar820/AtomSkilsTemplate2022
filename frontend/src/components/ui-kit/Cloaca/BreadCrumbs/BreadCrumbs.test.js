import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { matchers } from 'jest-emotion';
import BreadCrumbs from './';
import { testStyles } from 'lib-root/utils';

Enzyme.configure({ adapter: new Adapter() });
expect.extend(matchers);

const mockFn = jest.fn();

const props = {
    size: 'sm',
    items: [{ link: '#', title: '1' }, { title: '2' }, { link: '#', title: '2' }],
    lastItemColor: '#000',
    onLinkClick: mockFn,
    wrapStyle: { background: '#fff' },
    itemStyle: { background: '#000' },
    iconStyle: { background: '#eee' },
    width: '100px',
    height: '90px'
};

const Welcome = ({ link, title }) => <a href={link}>{title}</a>;

describe('<BreadCrumbs />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(<BreadCrumbs {...props} />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    test('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BreadCrumbs {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('should render correctly', () => {
        expect(wrapper.props()).toHaveProperty('size', props.size);
        expect(wrapper).toHaveStyleRule('font-size', '16px');

        expect(wrapper.props()).toHaveProperty('items', props.items);
        expect(wrapper.find('div').children()).toHaveLength(props.items.length);

        expect(wrapper.props()).toHaveProperty('wrapStyle', props.wrapStyle);
        testStyles(wrapper, props.wrapStyle);

        expect(wrapper.props()).toHaveProperty('itemStyle', props.itemStyle);
        testStyles(wrapper.find('StyledItem').at(0), props.itemStyle);

        expect(wrapper.props()).toHaveProperty('iconStyle', props.iconStyle);
        testStyles(wrapper.find('withColors(InlineIcons)').at(0), props.iconStyle);

        expect(wrapper.props()).toHaveProperty('width', props.width);
        expect(wrapper).toHaveStyleRule('width', props.width);

        expect(wrapper.props()).toHaveProperty('height', props.height);
        expect(wrapper).toHaveStyleRule('height', props.height);
    });

    test('should call onClick method', () => {
        wrapper
            .find('StyledLink')
            .at(0)
            .simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });

    test('should current lastItemColor', () => {
        expect(
            wrapper
                .find('withColors(InlineIcons)')
                .at(1)
                .prop('color')
        ).toEqual(props.lastItemColor);
    });

    test('should render disabled links', () => {
        expect(
            wrapper
                .find('StyledLink')
                .at(1)
                .prop('disabled')
        ).toBeTruthy();
    });

    test('should render correctly with linkComponent', () => {
        wrapper.setProps({
            linkComponent: <Welcome />
        });
        expect(wrapper.find('Welcome')).toHaveLength(props.items.length - 1);
    });

    test('should call onClick method with linkComponent', () => {
        wrapper.setProps({
            linkComponent: <Welcome />
        });
        wrapper
            .find('Welcome')
            .at(0)
            .simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });
});
