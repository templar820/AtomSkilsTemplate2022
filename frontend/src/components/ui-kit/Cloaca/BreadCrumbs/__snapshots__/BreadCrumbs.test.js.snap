// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadCrumbs /> should match snapshot 1`] = `
<BreadCrumbs
  height="90px"
  iconStyle={
    Object {
      "background": "#eee",
    }
  }
  itemStyle={
    Object {
      "background": "#000",
    }
  }
  items={
    Array [
      Object {
        "link": "#",
        "title": "1",
      },
      Object {
        "title": "2",
      },
      Object {
        "link": "#",
        "title": "2",
      },
    ]
  }
  lastItemColor="#000"
  onLinkClick={[MockFunction]}
  size="sm"
  width="100px"
  wrapStyle={
    Object {
      "background": "#fff",
    }
  }
>
  <Wrapper
    className="CLR-bread-crumbs"
    height="90px"
    size="sm"
    width="100px"
    wrapStyle={
      Object {
        "background": "#fff",
      }
    }
  >
    <div
      className="CLR-bread-crumbs css-1egxnzo-Wrapper e65b6qy0"
      height="90px"
      size="sm"
      width="100px"
    >
      <StyledItem
        className="CLR-bread-crumbs__item"
        color="#000"
        fontSize="0.8571428571428571rem"
        itemStyle={
          Object {
            "background": "#000",
          }
        }
        key="breadC_0"
      >
        <span
          className="CLR-bread-crumbs__item css-9tv2zt-StyledItem e65b6qy3"
          color="#000"
          fontSize="0.8571428571428571rem"
        >
          <StyledLink
            colors={
              Object {
                "GrayScale_0": "#ffffff",
                "GrayScale_100": "#EDEDED",
                "GrayScale_150": "#E5E5E5",
                "GrayScale_200": "#D8D8D8",
                "GrayScale_50": "#fafafa",
                "GrayScale_500": "#999999",
                "GrayScale_55": "#f6f6f6",
                "GrayScale_600": "#7C7C7C",
                "GrayScale_700": "#333333",
                "GrayScale_900": "#000000",
                "error": "#F62434",
                "errorAccent": "#EB5A40",
                "info": "#34AAF2",
                "infoAccent": "#5BD9E5",
                "primary": "#00B956",
                "primaryAccent": "#10E272",
                "secondary": "#731982",
                "secondaryAccent": "#A500BF",
                "tertiary": "#444189",
                "tertiaryAccent": "#404D46",
                "warning": "#FFA717",
                "warningAccent": "#F36F20",
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            <a
              className="css-1lpupas-StyledLink e65b6qy1"
              disabled={false}
              onClick={[Function]}
            >
              1
            </a>
          </StyledLink>
          <StyledIcon
            color="#333333"
            disabled={false}
            h="24px"
            height={null}
            icon="Arrow_type_1"
            iconStyle={
              Object {
                "background": "#eee",
              }
            }
            w="24px"
            width={null}
          >
            <withColors(InlineIcons)
              className="css-og27j6-StyledIcon e65b6qy4"
              color="#333333"
              disabled={false}
              h="24px"
              height={null}
              icon="Arrow_type_1"
              iconStyle={
                Object {
                  "background": "#eee",
                }
              }
              w="24px"
              width={null}
            >
              <InlineIcons
                className="css-og27j6-StyledIcon e65b6qy4"
                color="#333333"
                colors={
                  Object {
                    "GrayScale_0": "#ffffff",
                    "GrayScale_100": "#EDEDED",
                    "GrayScale_150": "#E5E5E5",
                    "GrayScale_200": "#D8D8D8",
                    "GrayScale_50": "#fafafa",
                    "GrayScale_500": "#999999",
                    "GrayScale_55": "#f6f6f6",
                    "GrayScale_600": "#7C7C7C",
                    "GrayScale_700": "#333333",
                    "GrayScale_900": "#000000",
                    "error": "#F62434",
                    "errorAccent": "#EB5A40",
                    "info": "#34AAF2",
                    "infoAccent": "#5BD9E5",
                    "primary": "#00B956",
                    "primaryAccent": "#10E272",
                    "secondary": "#731982",
                    "secondaryAccent": "#A500BF",
                    "tertiary": "#444189",
                    "tertiaryAccent": "#404D46",
                    "warning": "#FFA717",
                    "warningAccent": "#F36F20",
                  }
                }
                disabled={false}
                h="24px"
                height={null}
                icon="Arrow_type_1"
                iconStyle={
                  Object {
                    "background": "#eee",
                  }
                }
                w="24px"
                width={null}
              >
                <Styled(svg)
                  className="css-og27j6-StyledIcon e65b6qy4 CLR-inline-icons"
                  color="#333333"
                  colors={
                    Object {
                      "GrayScale_0": "#ffffff",
                      "GrayScale_100": "#EDEDED",
                      "GrayScale_150": "#E5E5E5",
                      "GrayScale_200": "#D8D8D8",
                      "GrayScale_50": "#fafafa",
                      "GrayScale_500": "#999999",
                      "GrayScale_55": "#f6f6f6",
                      "GrayScale_600": "#7C7C7C",
                      "GrayScale_700": "#333333",
                      "GrayScale_900": "#000000",
                      "error": "#F62434",
                      "errorAccent": "#EB5A40",
                      "info": "#34AAF2",
                      "infoAccent": "#5BD9E5",
                      "primary": "#00B956",
                      "primaryAccent": "#10E272",
                      "secondary": "#731982",
                      "secondaryAccent": "#A500BF",
                      "tertiary": "#444189",
                      "tertiaryAccent": "#404D46",
                      "warning": "#FFA717",
                      "warningAccent": "#F36F20",
                    }
                  }
                  disabled={false}
                  height="24px"
                  iconStyle={
                    Object {
                      "background": "#eee",
                    }
                  }
                  width="24px"
                >
                  <svg
                    className="e65b6qy4 CLR-inline-icons css-szh3dn-StyledIcon e1a7dme70"
                    color="#333333"
                    disabled={false}
                    height="24px"
                    width="24px"
                  >
                    <use
                      xlinkHref="#Arrow_type_1"
                    />
                  </svg>
                </Styled(svg)>
              </InlineIcons>
            </withColors(InlineIcons)>
          </StyledIcon>
        </span>
      </StyledItem>
      <StyledItem
        className="CLR-bread-crumbs__item"
        color="#333333"
        fontSize="0.8571428571428571rem"
        itemStyle={
          Object {
            "background": "#000",
          }
        }
        key="breadC_1"
      >
        <span
          className="CLR-bread-crumbs__item css-9tv2zt-StyledItem e65b6qy3"
          color="#333333"
          fontSize="0.8571428571428571rem"
        >
          <StyledLink
            colors={
              Object {
                "GrayScale_0": "#ffffff",
                "GrayScale_100": "#EDEDED",
                "GrayScale_150": "#E5E5E5",
                "GrayScale_200": "#D8D8D8",
                "GrayScale_50": "#fafafa",
                "GrayScale_500": "#999999",
                "GrayScale_55": "#f6f6f6",
                "GrayScale_600": "#7C7C7C",
                "GrayScale_700": "#333333",
                "GrayScale_900": "#000000",
                "error": "#F62434",
                "errorAccent": "#EB5A40",
                "info": "#34AAF2",
                "infoAccent": "#5BD9E5",
                "primary": "#00B956",
                "primaryAccent": "#10E272",
                "secondary": "#731982",
                "secondaryAccent": "#A500BF",
                "tertiary": "#444189",
                "tertiaryAccent": "#404D46",
                "warning": "#FFA717",
                "warningAccent": "#F36F20",
              }
            }
            disabled={true}
            onClick={[Function]}
          >
            <a
              className="css-1jv009j-StyledLink e65b6qy1"
              disabled={true}
              onClick={[Function]}
            >
              2
            </a>
          </StyledLink>
          <StyledIcon
            color="#000"
            disabled={true}
            h="24px"
            height={null}
            icon="Arrow_type_1"
            iconStyle={
              Object {
                "background": "#eee",
              }
            }
            w="24px"
            width={null}
          >
            <withColors(InlineIcons)
              className="css-og27j6-StyledIcon e65b6qy4"
              color="#000"
              disabled={true}
              h="24px"
              height={null}
              icon="Arrow_type_1"
              iconStyle={
                Object {
                  "background": "#eee",
                }
              }
              w="24px"
              width={null}
            >
              <InlineIcons
                className="css-og27j6-StyledIcon e65b6qy4"
                color="#000"
                colors={
                  Object {
                    "GrayScale_0": "#ffffff",
                    "GrayScale_100": "#EDEDED",
                    "GrayScale_150": "#E5E5E5",
                    "GrayScale_200": "#D8D8D8",
                    "GrayScale_50": "#fafafa",
                    "GrayScale_500": "#999999",
                    "GrayScale_55": "#f6f6f6",
                    "GrayScale_600": "#7C7C7C",
                    "GrayScale_700": "#333333",
                    "GrayScale_900": "#000000",
                    "error": "#F62434",
                    "errorAccent": "#EB5A40",
                    "info": "#34AAF2",
                    "infoAccent": "#5BD9E5",
                    "primary": "#00B956",
                    "primaryAccent": "#10E272",
                    "secondary": "#731982",
                    "secondaryAccent": "#A500BF",
                    "tertiary": "#444189",
                    "tertiaryAccent": "#404D46",
                    "warning": "#FFA717",
                    "warningAccent": "#F36F20",
                  }
                }
                disabled={true}
                h="24px"
                height={null}
                icon="Arrow_type_1"
                iconStyle={
                  Object {
                    "background": "#eee",
                  }
                }
                w="24px"
                width={null}
              >
                <Styled(svg)
                  className="css-og27j6-StyledIcon e65b6qy4 CLR-inline-icons"
                  color="#000"
                  colors={
                    Object {
                      "GrayScale_0": "#ffffff",
                      "GrayScale_100": "#EDEDED",
                      "GrayScale_150": "#E5E5E5",
                      "GrayScale_200": "#D8D8D8",
                      "GrayScale_50": "#fafafa",
                      "GrayScale_500": "#999999",
                      "GrayScale_55": "#f6f6f6",
                      "GrayScale_600": "#7C7C7C",
                      "GrayScale_700": "#333333",
                      "GrayScale_900": "#000000",
                      "error": "#F62434",
                      "errorAccent": "#EB5A40",
                      "info": "#34AAF2",
                      "infoAccent": "#5BD9E5",
                      "primary": "#00B956",
                      "primaryAccent": "#10E272",
                      "secondary": "#731982",
                      "secondaryAccent": "#A500BF",
                      "tertiary": "#444189",
                      "tertiaryAccent": "#404D46",
                      "warning": "#FFA717",
                      "warningAccent": "#F36F20",
                    }
                  }
                  disabled={true}
                  height="24px"
                  iconStyle={
                    Object {
                      "background": "#eee",
                    }
                  }
                  width="24px"
                >
                  <svg
                    className="e65b6qy4 CLR-inline-icons css-1yecxrn-StyledIcon e1a7dme70"
                    color="#000"
                    disabled={true}
                    height="24px"
                    width="24px"
                  >
                    <use
                      xlinkHref="#Arrow_type_1"
                    />
                  </svg>
                </Styled(svg)>
              </InlineIcons>
            </withColors(InlineIcons)>
          </StyledIcon>
        </span>
      </StyledItem>
      <StyledItem
        className="CLR-bread-crumbs__item"
        color="#333333"
        fontSize="0.8571428571428571rem"
        itemStyle={
          Object {
            "background": "#000",
          }
        }
        key="breadC_2"
      >
        <span
          className="CLR-bread-crumbs__item css-9tv2zt-StyledItem e65b6qy3"
          color="#333333"
          fontSize="0.8571428571428571rem"
        >
          <StyledText
            fontSize="0.8571428571428571rem"
            lastItemColor="#000"
          >
            <span
              className="css-1hladgy-StyledText e65b6qy2"
              fontSize="0.8571428571428571rem"
            >
              2
            </span>
          </StyledText>
        </span>
      </StyledItem>
    </div>
  </Wrapper>
</BreadCrumbs>
`;
